### PR TITLE
Add load more button e2e tests

### LINKED ISSUE #1788: Add e2e tests for Load more functionality
## Problem

Load more functionality needs a suite of e2e tests. The cases should use both SSR and CSR to load the first search page.

 1. All content view with more than 1 page of results for each media type:
  - should have a Load more button.
  - each individual media type view should have a Load more button.
  - when button is clicked, should fetch all media types.
 2. All content view with results for images, but only 1 page of results for audio:
  - should have a Load more button.
  - when button is clicked, should fetch only images.
  - image view should have a Load more button, but audio view should not.
 3. All content view with results for images, but no results for audio:
  - should have a Load more button.
  - when button is clicked, should fetch only images.
  - image view should have a Load more button.

## Implementation

- [ ] 🙋 I would be interested in implementing this feature.

### PR DESCRIPTION
**Pull request by obulat**
## Fixes

Fixes #1788 by @obulat

## Description

This PR adds the e2e tests for load more functionality. They should fail on main, but pass after #1786 is merged.

## Checklist

- [x] My pull request has a descriptive title (not a vague title like `Update index.md`).
- [x] My pull request targets the *default* branch of the repository (`main`) or a parent feature branch.
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [x] I added or updated tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no visible errors.

[best_practices]:https://git-scm.com/book/en/v2/Distributed-Git-Contributing-to-a-Project#_commit_guidelines

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: test/playwright/e2e/load-more.spec.ts [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+ *  - when button is clicked, should fetch all media types.
+ * 2. All content view with results for images, but only 1 page of results for audio:
+ *  - should have a Load more button.
+ *  - when button is clicked, should fetch only images.
+ *  - image view should have a Load more button, but audio view should not.
+ * 3. All content view with results for images, but no results for audio:
+ *  - should have a Load more button.
+ *  - when button is clicked, should fetch only images.
+ *  - image view should have a Load more button.
+ */
```
Comment (dhruvkb) [inline test/playwright/e2e/load-more.spec.ts:44]: Although it is unlikely to differ in behaviour, but for completeness' sake, we should add the opposite of cases 2 and 3 where audio results are more than image.

**Review by dhruvkb — APPROVED** [review]
LGTM! These tests inspire confidence that future issues will be caught during testing.

File: test/playwright/e2e/load-more.spec.ts (hunk 1 above)
Comment (obulat) [reply test/playwright/e2e/load-more.spec.ts:44]: I agree, but I couldn't find an example search term for that 😆 I guess I should mock the response.

Comment (zackkrida) [inline test/playwright/e2e/load-more.spec.ts:44]: Fantastic!

File: test/playwright/e2e/load-more.spec.ts [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+        await openSingleMediaView(page, IMAGE)
+        await expect(page.locator(loadMoreButton)).toBeVisible()
+
+        await page.goBack()
+
+        await openSingleMediaView(page, AUDIO)
+        await expect(page.locator(loadMoreButton)).toBeVisible()
+      })
+
+      test(`Renders on All view when images have results but audio don't`, async ({
```
Comment (zackkrida) [inline test/playwright/e2e/load-more.spec.ts:87]:
```suggestion
      test(`Renders on All view when images have results but audio does not`, async ({
```

small grammatical nit

File: test/playwright/e2e/load-more.spec.ts [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+          const url = req.url()
+          if (url.includes('page=2')) {
+            if (url.includes('/audio/')) {
+              additionalRequests.push(AUDIO)
+            } else if (url.includes('/images/')) {
+              additionalRequests.push(IMAGE)
+            }
+          }
+        })
+        await goToSearchTerm(page, 'horses snort', { mode })
```
Comment (zackkrida) [inline test/playwright/e2e/load-more.spec.ts:112]: > horses snort

😆

**Review by zackkrida — APPROVED** [review]
LGTM

**Pull request merged**